docs/index.rst

```diff
@@ -384,6 +384,21 @@ dynamically loaded plugins.
 For more info see :ref:`call_historic`.
 
 
+Warnings on hook implementation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+As projects evolve new hooks may be introduced and/or deprecated.
+
+if a hookspec specifies a impl_warning, pluggy will trigger it for any plugin implementing the hook.
+
+
+.. code-block:: python
+
+    @hookspec(impl_warning=DeprecationWarning("oldhook is deprecated and will be removed soon"))
```

Comment:
What about just `warning`? 
I don't know that the `impl_` adds anything useful.

Reply:
warning alone is a overly generic name, thats why i added the impl in front - im happy about picking a more specific name thats describing what is happening, im severely opposed to going for a more generic name

its probably a good idea to brainstorm a better name, its fine if its a bit longer

Reply:
It might be *general* but it's as specific as the code you've written. 

The *type* of warning is handed into the decorator and the underlying code merely uses the `warnings` mod to `warn`. Unless you intend to limit to specific subset of [warning categories](https://docs.python.org/3.6/library/warnings.html#warning-categories) (which doesn't seem all that likely nor easy to describe in a variable name) I don't see how it can be more specific.

Reply:
@tgoodlet im not talking about types or calls - im talking about the intent of the value passed - which is a warning that is given for each implementation of the hook encountered - its bad style to encode types, its good style to enode intent

Reply:
Ok what about `warning_on_call`?

Reply:
the warnings are happening at registration time, not invocation time, so on call would be wrong

`warning_on_impl` or `warning_on_implementation` might be more to the point tho

Reply:
`warn_on_register`?

Reply:
i'll sleep a night over it and see what the impression is

Reply:
i decided to call it warn_on_impl since we consistently use **impl** as shortcut for implementation